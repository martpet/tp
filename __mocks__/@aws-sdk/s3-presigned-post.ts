export const createPresignedPost = vi.fn().mockResolvedValue('dummyPresignedPost');
