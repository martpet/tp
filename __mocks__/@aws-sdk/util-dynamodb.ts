export const unmarshall = vi
  .fn()
  .mockName('unmarshall')
  .mockReturnValue('dummyUnmarshalledItems');
