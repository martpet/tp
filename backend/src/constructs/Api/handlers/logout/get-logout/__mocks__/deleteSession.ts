export const deleteSession = vi.fn().mockResolvedValue('dummyRefreshToken');
