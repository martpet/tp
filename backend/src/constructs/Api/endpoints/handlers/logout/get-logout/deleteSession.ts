import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
import { DeleteCommand, DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';

import { sessionsTableOptions } from '~/consts';
import { SessionsTableItem } from '~/types';

const ddbClient = new DynamoDBClient({});
const ddbDocClient = DynamoDBDocumentClient.from(ddbClient);

export const deleteSession = async (sessionId: string) => {
  const deleteCommand = new DeleteCommand({
    TableName: sessionsTableOptions.tableName,
    Key: { [sessionsTableOptions.partitionKey.name]: sessionId },
    ReturnValues: 'ALL_OLD',
  });

  const { Attributes } = await ddbDocClient.send(deleteCommand);
  const sessionsItem = Attributes as SessionsTableItem | undefined;

  if (!sessionsItem) {
    throw new Error(`Could not find "Sessions" item with id "${sessionId}"`);
  }

  return sessionsItem.refreshToken;
};
