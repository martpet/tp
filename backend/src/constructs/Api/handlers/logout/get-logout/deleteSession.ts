import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
import { DeleteCommand, DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';

import { sessionsTableOptions } from '~/consts';
import { SessionsTableItem } from '~/types';

const ddbClient = new DynamoDBClient({});
const ddbDocumentClient = DynamoDBDocumentClient.from(ddbClient);

export const deleteSession = async (sessionId: string) => {
  const deleteCommand = new DeleteCommand({
    TableName: sessionsTableOptions.tableName,
    Key: { [sessionsTableOptions.partitionKey.name]: sessionId },
    ReturnValues: 'ALL_OLD',
  });

  const { Attributes } = await ddbDocumentClient.send(deleteCommand);
  const item = Attributes as SessionsTableItem | undefined;

  return item!.refreshToken;
};
