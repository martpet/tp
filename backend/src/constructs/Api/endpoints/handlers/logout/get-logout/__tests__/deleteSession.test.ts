import { DeleteCommand, DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
import { mockClient } from 'aws-sdk-client-mock';

import { deleteSession } from '../deleteSession';

const ddbMock = mockClient(DynamoDBDocumentClient);

const args = ['dummySessionId'] as const;

beforeEach(() => {
  ddbMock.reset();

  ddbMock
    .on(DeleteCommand)
    .resolves({ Attributes: { refreshToken: 'dummyRefreshToken' } });
});

describe('deleteSession', () => {
  it('sends "DeleteCommand" to DynamoDB with correct args', async () => {
    await deleteSession(...args);
    expect(ddbMock.commandCalls(DeleteCommand)[0].args[0].input).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    return expect(deleteSession(...args)).resolves.toMatchSnapshot();
  });

  describe('when "Attributes" prop is missing from "DeleteCommand" output', () => {
    beforeEach(() => {
      ddbMock.on(DeleteCommand).resolves({});
    });

    it('rejects with a correct value', () => {
      return expect(deleteSession(...args)).rejects.toMatchSnapshot();
    });
  });
});
