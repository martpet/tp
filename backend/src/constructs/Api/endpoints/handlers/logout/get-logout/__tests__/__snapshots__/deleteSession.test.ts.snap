// Vitest Snapshot v1

exports[`deleteSession > resolves with a correct value 1`] = `"dummyRefreshToken"`;

exports[`deleteSession > sends "DeleteCommand" to DynamoDB with correct args 1`] = `
{
  "Key": {
    "id": "dummySessionId",
  },
  "ReturnValues": "ALL_OLD",
  "TableName": "TripPics-Sessions",
}
`;

exports[`deleteSession > when "Attributes" prop is missing from "DeleteCommand" output > rejects with a correct value 1`] = `[Error: Could not find "Sessions" item with id "dummySessionId"]`;
